src/universal/generate-random-pkmn.c

```diff
@@ -0,0 +1,155 @@
+#include <pksm.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <time.h> 
+#include <string.h>
+
+int main(int argc, char **argv)
+{
+    unsigned char version = *argv[2];
+    enum Generation gen;
+
+    int maxSpecies = 0;
+    int maxMoves = 0;
+    int maxBalls = 0;
+    int maxItems = 0;
+    int maxAbility = 0;
+
+    gui_warn("Please enter how many pokemon", "you would like to generate!");
+
+    int randomizedCount = 0;
+    gui_numpad(&randomizedCount, "You are entering how many randomized\npokemon you would like to generate", 3);
+
+    switch (version)
+    {
+        case 10:
+        case 11:
+        case 12:
+        case 7:
+        case 8:
+            gen = GEN_FOUR;
+            maxSpecies = 493;
+            maxBalls = 18;
+            maxMoves = 467;
+            if(version == 11 || version == 10){
```

Comment:
Why not just do these in the cases themselves?

Reply:
I want to keep all the gens grouped together.

Reply:
Fair enough